janusgraph-core/src/main/java/org/janusgraph/graphdb/configuration/GraphDatabaseConfiguration.java

```diff
@@ -292,6 +293,18 @@ public boolean apply(@Nullable String s) {
                     "default", JanusGraphDefaultSchemaMaker.INSTANCE,
                     "tp3", Tp3DefaultSchemaMaker.INSTANCE);
 
+    public static final ConfigOption<String> MAPPING_HANDLING = new ConfigOption<>(SCHEMA_NS, "mapping",
+        "Configures the behavior of the schema handling. If set to 'none', mapping constraints are not applied. " +
+            "If set to 'constraints', mapping constraints are applied",
+        ConfigOption.Type.MASKABLE, "none", new Predicate<String>() {
+        @Override
+        public boolean apply(@Nullable String s) {
+            return s != null && preregisteredMappingHandling.containsKey(s);
+        }
+    });
+
+    private static final Map<String, Boolean> preregisteredMappingHandling =
```

Comment:
`preregisteredMappingHandling` -> `preregisteredSchemaMode` (I won't repeat this for the other occurrences of _mapping_.)